### PR TITLE
[Bug] Add exists check for IOCs index.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Added check to prevent resource_already_exists_exception when indexing more than 10k iocs.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security-analytics/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
Adding tests locally currently.

File: src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java [hunk 1]
```diff
@@ -250,6 +252,11 @@ private void initFeedIndex(String feedIndexName, ActionListener<CreateIndexRespo
                         listener.onResponse(r);
                     },
                     e -> {
+                        if (e instanceof ResourceAlreadyExistsException || (e instanceof RemoteTransportException && e.getCause() instanceof ResourceAlreadyExistsException)) {
+                            log.debug("index {} already exist", iocIndexMapping());
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java:256]: shouldnt this variable be `feedIndexName`?

Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java:256]: I had based it on this function, I can change it to the index name though if we prefer.
https://github.com/opensearch-project/security-analytics/blob/main/src/main/java/org/opensearch/securityanalytics/threatIntel/iocscan/dao/IocFindingService.java#L44

Comment (AWSHurneyt) [reply src/main/java/org/opensearch/securityanalytics/services/STIX2IOCFeedStore.java]: Revised.

**Review by eirsep — APPROVED** [review]

**Review by jowg-amazon — APPROVED** [review]

**Pull request merged**